prow/jira/jira.go

```diff
@@ -107,8 +100,33 @@ func NewClient(endpoint string, opts ...Option) (Client, error) {
 		}
 	}
 
-	jiraClient, err := jira.NewClient(retryingClient.StandardClient(), endpoint)
-	return &client{upstream: jiraClient, url: endpoint}, err
+	return jira.NewClient(retryingClient.StandardClient(), endpoint)
+}
+
+func NewClient(endpoint string, opts ...Option) (Client, error) {
+	o := Options{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+
+	log := logrus.WithField("client", "jira")
+	if len(o.LogFields) > 0 {
+		log = log.WithFields(o.LogFields)
+	}
+	retryingClient := retryablehttp.NewClient()
+	usedFlagTransport := &clientUsedTransport{
+		m:        sync.Mutex{},
+		upstream: retryingClient.HTTPClient.Transport,
+	}
+	retryingClient.HTTPClient.Transport = usedFlagTransport
+	retryingClient.Logger = &retryableHTTPLogrusWrapper{log: log}
+
+	var url url.URL
+	jiraClient, err := newJiraClient(endpoint, o, retryingClient)
+	if err != nil {
+		url = jiraClient.GetBaseURL()
```

Comment:
We can not access a non-err return if we had an error, this needs to be an `url.Parse(endpoint)` or some such and also consider the possibility that endpoint might not be a valid url

Reply:
This point I don't get. Can you rephrase?

Reply:
if `newJiraClient` returns a non-nil error, you can now access `jiraClient`, because whenever you get an `err` from a function, you can not make any assumption about the other returns of that function. Here, we assume that even in the error case it returns a non-nil `jiraClient`